src/vs/editor/common/languages.ts

```diff
@@ -176,11 +192,33 @@ export interface Hover {
  */
 export interface HoverProvider {
```

Comment:
I think you need some `free(hoverId: string): void` / `free(hover: THover): void` to properly implement this.